Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs

```diff
@@ -0,0 +1,176 @@
+        var invComp = SEntMan.GetComponent<InventoryComponent>(SPlayer);
+
+        var actionSys = Server.System<ActionsSystem>();
+        var baseActions = actionSys.GetActions(SPlayer).ToList();
+
+        foreach (var clothing in clothings)
+        {
+            Assert.That(SEntMan.TryGetComponent<ClothingComponent>(clothing, out var clothingComp),
+                $"{SEntMan.ToPrettyString(clothing)} doesn't have ClothingComponent");
+            foreach (var slot in invComp.Slots)
+            {
+                if ((slot.SlotFlags & clothingComp!.Slots) == 0x0)
+                    continue;
+
+                await Server.WaitAssertion(() =>
+                {
+                    if (slot.SlotFlags is SlotFlags.IDCARD or SlotFlags.POCKET)
+                        DoEquip(invSystem, jumpsuit, "jumpsuit");
+
+                    if (slot.SlotFlags == SlotFlags.SUITSTORAGE)
+                        DoEquip(invSystem, hardsuit, "outerClothing");
+
+                    DoEquip(invSystem, clothing, slot.Name, clothingComp);
+                });
+                await RunTicks(1);
+
+                var newActions = actionSys.GetActions(SPlayer)
+                    .Except(baseActions)
+                    // Don't bother with more complicated actions (they can take targets, etc.)
+                    .Where(action => SEntMan.HasComponent<InstantActionComponent>(action.Owner))
+                    .ToList();
+
+                await Server.WaitPost(() => RunActions(newActions, actionSys));
+                await RunTicks(1);
+
+                var unremovableComp = SEntMan.GetComponentOrNull<UnremoveableComponent>(clothing);
+                // Electropack or similar has to be force removed, since our unequip actor is always ourselves
+                var needsForce = SEntMan.HasComponent<SelfUnremovableClothingComponent>(clothing);
+
+                if (unremovableComp != null)
+                {
+                    needsForce = true;
+                    // We want to reuse this entity for other slots. (E.g., CluwnePDA)
+                    unremovableComp.DeleteOnDrop = false;
+                }
+
+                var isRemoved = false;
+                await Server.WaitPost(() =>
+                {
+                    // Try without forcing first
+                    isRemoved = DoUnequip(invSystem, clothing, slot.Name, false, clothingComp, false);
+
+                    // If we should've needed to force for this item, but got it off, that's no good.
+                    Assert.That(isRemoved && needsForce,
+                        Is.False,
+                        $"Unequipped {SEntMan.ToPrettyString(clothing)}, which should have been unremovable.");
+
+                    // Just in case some kind of toggle action put us into a state that this test doesn't expect.
+                    // Currently this doesn't seem to get hit.
+                    if (!isRemoved)
+                        RunActions(newActions, actionSys);
```

Comment:
Should this be removed? In the original implementation I thought it was needed, but I'm not seeing any cases where this is a behavior that exists.